operations/mimir/ingest-storage-migration.libsonnet

```diff
@@ -0,0 +1,216 @@
+{
+  local statefulSet = $.apps.v1.statefulSet,
+
+  _config+:: {
+    // Controls whether ingester-zone-X-partition should be deployed even if ingest storage is not enabled globally.
+    ingest_storage_migration_partition_ingester_zone_a_enabled: false,
+    ingest_storage_migration_partition_ingester_zone_b_enabled: false,
+    ingest_storage_migration_partition_ingester_zone_c_enabled: false,
+
+    // Controls whether write path components should tee writes both to classic ingesters and Kafka.
+    ingest_storage_migration_write_to_partition_ingesters_enabled: false,
+    ingest_storage_migration_write_to_classic_ingesters_enabled: false,
+
+    // Controls whether reads path components should read from partition ingesters instead of classic ones.
+    ingest_storage_migration_querier_enabled: false,
+
+    // Controls the procedure to decommission classic ingesters.
+    ingest_storage_migration_classic_ingesters_scale_down: false,
+    ingest_storage_migration_classic_ingesters_zone_a_decommission: false,
+    ingest_storage_migration_classic_ingesters_zone_b_decommission: false,
+    ingest_storage_migration_classic_ingesters_zone_c_decommission: false,
+
+    // Controls the procedure to decommission partition ingesters (to rename them back like classic ones).
+    ingest_storage_migration_partition_ingester_zone_a_scale_down: false,
+    ingest_storage_migration_partition_ingester_zone_b_scale_down: false,
+    ingest_storage_migration_partition_ingester_zone_c_scale_down: false,
+  },
+
+  //
+  // Partition ingesters (temporarily deployment used during the migration).
+  //
+  // Notes:
+  // - To keep jsonnet changes easier, classic and partition ingesters share the same rollout-group. This means that,
+  //   from the rollout-operator perspective, classic and partition ingesters are the same group and their StatefulSets
+  //   get rolled out sequentially (even if classic and partition ingesters could be rolled out concurrently).
+  // - No PodDisruptionBudget for partition ingesters because "ingester-rollout" PDB also covers partition ingesters,
+  //   since the rollout-group is shared between the two.
+  //
+
+  local partitionIngesterArgs =
+    // Explicitly include all the ingest storage config because during the migration the ingest storage
+    // may not be enabled globally yet.
+    $.ingest_storage_args +
+    $.ingest_storage_kafka_consumer_args +
+    $.ingest_storage_ingester_args +
+    $.ingest_storage_ingester_ring_client_args + {
+      // Run partition ingesters on a dedicated hash ring, so that they don't clash with classic ingesters.
+      'ingester.ring.prefix': 'partition-ingesters/',
```

Comment:
nit: Would be nice to use common local var for `partition-ingesters/`.

Reply:
Done in https://github.com/grafana/mimir/pull/8028/commits/b02e79db8450f4dd7df3346097aed6528248fbfa. I've used a config option instead of a local var, because it's scattered around 2 files.